clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreAwarePartitionWiseKafkaConsumerService.java

```diff
@@ -0,0 +1,200 @@
+package com.linkedin.davinci.kafka.consumer;
+
+import com.linkedin.davinci.stats.AggKafkaConsumerServiceStats;
+import com.linkedin.venice.meta.ReadOnlyStoreRepository;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.pubsub.PubSubConsumerAdapterFactory;
+import com.linkedin.venice.pubsub.api.PubSubConsumerAdapter;
+import com.linkedin.venice.pubsub.api.PubSubMessageDeserializer;
+import com.linkedin.venice.pubsub.api.PubSubTopic;
+import com.linkedin.venice.pubsub.api.PubSubTopicPartition;
+import com.linkedin.venice.utils.Time;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import io.tehuti.metrics.MetricsRepository;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import org.apache.logging.log4j.LogManager;
+
+
+/**
+ * {@link StoreAwarePartitionWiseKafkaConsumerService} is used to allocate share consumer from consumer pool at partition
+ * granularity. One shared consumer may have multiple topics, and each topic may have multiple consumers.
+ * This is store-aware version of topic-wise shared consumer service. The topic partition assignment in this service has
+ * a heuristic that we should distribute the all the subscriptions related to a same store / version as even as possible.
+ * The load calculation for each consumer will be:
+ * Hybrid store: Consumer assignment size + STORE_SUBSCRIPTION_LOAD * subscription count for the same store;
+ * Batch only store: Consumer assignment size + STORE_SUBSCRIPTION_LOAD * subscription count for the same store version;
+ * and we will pick the least loaded consumer for a new topic partition request.
+ */
+public class StoreAwarePartitionWiseKafkaConsumerService extends PartitionWiseKafkaConsumerService {
+  private static final long DEFAULT_STORE_SUBSCRIPTION_LOAD = 100;
+  private final Map<String, Map<PubSubConsumerAdapter, Integer>> resourceIdentifierToConsumerMap =
```

Comment:
The naming of this field is not good enough to describe the purpose, so can you leave some javadoc?